src/makeInfinityAdder.js

```diff
@@ -5,8 +5,24 @@
  * @return {function}
  */
 
-function makeInfinityAdder() {
-  // write code here
+function makeInfinityAdder(n) {
+  let result = 0;
+  let zm = 0;
+
+  const add = (value) => {
+    if (value !== undefined) {
+      zm += value;
+
+      return add;
+    } else {
+      result = zm;
+      zm = 0;
+
+      return result;
+    }
```

Comment:
Else statement is useless here
```suggestion
    } 
    
    result = zm;
    zm = 0;

    return result;
```